portico: Reduce header font size to 11px on small screens.

amanagr opened the pull request:
This allows us fit the navbar links in the header, which avoids
them overflowing into next line.

The numbers were calculated on a hit and trial basis.

amanagr committed:
portico: Add hamburger menu for navbar links on mobile widths.

amanagr commented:
before:
<img width="825" alt="Screenshot 2021-10-07 at 11 29 47 AM" src="https://user-images.githubusercontent.com/25124304/136328139-bc72b39f-7121-4f7b-9c3f-5ab13fec32e4.png">

after:
<img width="825" alt="Screenshot 2021-10-07 at 11 29 27 AM" src="https://user-images.githubusercontent.com/25124304/136328165-280a6be8-adcc-4d47-8934-73ecef2150f3.png">

timabbott commented:
@alya thoughts on this?  I kinda think we might be better off putting those items inside a hamburger menu than shrinking them, especially since the strings could be longer when internationalized.

amanagr committed:
portico_signin: Fix display for mobile width devices.

Fix display of pages containing white-box on mobile widths.
Reduced width of buttons and forms on smaller widths to fit
the viewport.

The elements were misaligned on widths less than 360px. /login
could still be improved a bit but not sure what's increasing
its width out of the viewport on 320px.

amanagr commented:
<img width="438" alt="Screenshot 2021-11-22 at 7 51 27 PM" src="https://user-images.githubusercontent.com/25124304/142877819-189e6a2a-87ac-4418-b085-adb56af40ce4.png">
<img width="438" alt="Screenshot 2021-11-22 at 6 45 18 PM" src="https://user-images.githubusercontent.com/25124304/142877490-16072221-ca53-42b5-8a0d-9978d68b219f.png">

amanagr commented:
I implemented a basic version of the hamburger myself since the bootstrap version can no longer be used.

timabbott commented:
did you consider https://fontawesome.com/v4.7/icon/bars?

amanagr committed:
hello: Use fa-bars for hamburger icon.

amanagr commented:
Ok, replaced the hamburger on both pages.

alya commented:
Thanks! A couple of comments:

1) I don't think the narrow menu looks great right now... Could we use the same style as on the landing page instead?
![Screen Shot 2021-11-23 at 9 36 12 AM](https://user-images.githubusercontent.com/2090066/143075733-f079881c-14c2-49af-9fcc-8567b2486d63.png)

2) As long as we are working on this page, can we make sure this box resizes early enough not to get cut off?
![Screen Shot 2021-11-23 at 9 34 11 AM](https://user-images.githubusercontent.com/2090066/143075860-edfc6814-8163-4c3a-81e9-5bd3d7aa579d.png)
e

amanagr commented:
> I don't think the narrow menu looks great right now... Could we use the same style as on the landing page instead?

That is actually not made in a reusable fashion. I will have to redo the work of making it. @eeshangarg maybe efficient at doing so.

alya commented:
Ah, I see. @eeshangarg could you help out on this one?

eeshangarg commented:
@alya Will take a look tomorrow :)

eeshangarg commented:
@alya: I just spent some time playing around with this. @amanagr is right, the code for this has not been written in a re-usable way. So I will have to move some code around, but I think I can get this done.

@amanagr Would it be okay with you if I took over this PR?

amanagr commented:
Yup.

timabbott commented:
@eeshangarg ping on this -- I'd like to make sure that this highly visible bug in our mobile navigation doesn't get lost.

alya commented:
@amanagr does this PR address the page getting cut off on mobile (see comment above)? Google is starting to complain about it, so we definitely need to get that fixed.

For the menus, at this point I'm fine with just merging a version that is not broken, even if it's not perfect.

amanagr commented:
The current state of the PR is good to fix the existing issue. We can iterate on it later.

alya commented:
Great, thanks!

amanagr commented:
No longer reproduciable.
